### PR TITLE
list: add LIST_POP() helper that pops the first item off a linked list

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by poettering** [comment]
Split out of #20156.

File: src/basic/list.h [hunk 1]
```diff
@@ -181,3 +181,12 @@
                 }                                                       \
                 (b) = NULL;                                             \
         } while (false)
+
+#define LIST_POP(name, a)                                               \
+        ({                                                              \
+                typeof(a)* _a = &(a);                                   \
+                typeof(a) _p = *_a;                                     \
```
Comment (daandemeyer) [inline src/basic/list.h:188]: Is this done to avoid the double evaluation?

**Review by DaanDeMeyer — APPROVED** [review]

File: src/basic/list.h (hunk 1 above)
Comment (poettering) [reply src/basic/list.h:188]: yes, precisely, it's a common pattern in macros like this

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**